layouts/private-wrapper.tsx

```diff
@@ -11,12 +11,13 @@ interface PrivateWrapperProps {
 const PrivateWrapper = ({ isPrivateRoute = false, children }: PrivateWrapperProps) => {
   const user = useUser();
   const router = useRouter();
+  const { isLoading } = useSessionContext();
   const { session: isValid } = useSession(true);
 
   async function checkSession() {
     if (router.asPath?.includes("login")) return;
 
-    if (!isValid) {
+    if (!isValid && !isLoading) {
```

Comment:
Does this need to be in the `useEffect` deps array also?

Reply:
Yes, it would be beneficial to include it in the `useEffect` dependencies array to ensure the effect behaves as expected. Thank you for considering this!